pkgs/coverage/lib/src/coverage_percentage.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:math';
+
+import 'hitmap.dart';
+
+/// Calculates the coverage percentage from a hitmap.
+///
+/// [hitmap] is the map of file paths to hit maps.
+/// [precision] is the number of decimal places to round to (default: 0).
+///
+/// Returns a [CoverageResult] containing the coverage percentage and line
+/// counts.
+CoverageResult calculateCoveragePercentage(
+  Map<String, HitMap> hitmap, {
+  int precision = 0,
+}) {
+  var totalLines = 0;
+  var coveredLines = 0;
+
+  for (final entry in hitmap.entries) {
+    final lineHits = entry.value.lineHits;
```

Comment:
Can you also include the `branchHits`, if any? It's possible to have 100% line coverage but miss some branches. No need to include the `funcHits` as they are a subset of the `lineHits`.